logwatch/logwatch.go

```diff
@@ -0,0 +1,181 @@
+package logwatch
+
+import (
+	"context"
+	"regexp"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/prometheus/common/model"
+	"github.com/rs/zerolog"
+	"github.com/smartcontractkit/wasp"
+	"github.com/testcontainers/testcontainers-go"
+
+	"github.com/smartcontractkit/chainlink-testing-framework/utils"
+)
+
+// LogNotification notification about log line match for some container
+type LogNotification struct {
+	Container string
+	Log       string
+}
+
+// LogWatch is a test helper struct to monitor docker container logs for some patterns
+// and push their logs into Loki for further analysis
+type LogWatch struct {
+	t          *testing.T
+	log        zerolog.Logger
+	loki       *wasp.LokiClient
+	patterns   map[string][]*regexp.Regexp
+	notifyTest chan *LogNotification
+	containers []testcontainers.Container
+	consumers  map[string]*ContainerLogConsumer
+}
+
+// NewLogWatch creates a new LogWatch instance, with a Loki client
+func NewLogWatch(t *testing.T, patterns map[string][]*regexp.Regexp) (*LogWatch, error) {
+	loki, err := wasp.NewLokiClient(wasp.NewEnvLokiConfig())
+	if err != nil {
+		return nil, err
+	}
+	return &LogWatch{
+		t:          t,
+		log:        utils.GetTestComponentLogger(t, "LogWatch"),
+		loki:       loki,
+		patterns:   patterns,
+		notifyTest: make(chan *LogNotification, 10000),
+		consumers:  make(map[string]*ContainerLogConsumer, 0),
+	}, nil
+}
+
+// Listen listen for the next notification
+func (m *LogWatch) Listen() *LogNotification {
+	msg := <-m.notifyTest
+	m.log.Warn().
+		Str("Container", msg.Container).
+		Str("Line", msg.Log).
+		Msg("Received notification from container")
+	return msg
+}
+
+// OnMatch calling your testing hook on first match
+func (m *LogWatch) OnMatch(f func()) {
+	go func() {
+		for {
+			msg := <-m.notifyTest
+			m.log.Warn().
```

Comment:
Why is this a warn log?

Reply:
Usually, we'll search for specific or common panics/fatals, etc. Since it's not our `error`, `warn` seems ok.